### DISCUSSION TITLE
Access Audio Stream

### DISCUSSION THREAD

**Discussion by steckes** [mciantyre/teensy4-rs#145] [Q&A] [answered]
Hey I'm new to embedded rust programming and have a teensy 4.1 with audio boards.
Does anyone know how to access the audio stream of the teensy input and output in rust?

**Comment by mciantyre** [comment, marked as answer]
Hi there. From my understanding, the Teensy 4 audio boards use I2S signals for audio data. We don't have an I2S driver; see #141 for the last discussion.

The only update I can provide: there's a user in [the imxrt-rs matrix chat](https://matrix.to/#/#imxrt-rs:matrix.org) who's exploring a driver. If you're interested in driver development, there might be someone available for collaboration.
